### PR TITLE
add css option - fixes #24

### LINKED ISSUE #24: CSS support
In order to fix https://github.com/sindresorhus/pageres/issues/228, the first step would be to allow `screenshot-stream` to accept raw css.

I'm not very familiar with PhantomJS, but from the documentation of their website, it looks like there is no option to add custom css. So probably the best that could be done is by manually adding a style tag in the head of the page. Wonder if the following would do the trick if it was added in [page.evaluate](https://github.com/kevva/screenshot-stream/blob/master/stream.js#L70-L78)

```js
var style = document.createElement('style');
style.type = 'text/css';
style.appendChild(document.createTextNode(opts.css));

document.head.appendChild(style);
```

Will create a PR for this.

### PR DESCRIPTION
**Pull request by SamVerschueren**
Added support for a `css` option and it seems to work well. Feedback is more then welcome.

I am not sure if `pageres` should support a css file, but I think it would be a nice addition. Should `screenshot-stream` support a css file as well then? Or should `pageres` detect if it's a file, read the css file and pass the content of that file directly to `screenshot-stream`?

// @sindresorhus

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ Type: `string`
 
 Capture a specific DOM element.
 
+##### css
+
+Type: `string`
+
+Apply CSS styles to the webpage.
+
```
Comment (SamVerschueren) [inline readme.md]: Added this block here, not sure if it should be added at the bottom of the options list? Not sure about the description either.

**Comment by kevva** [comment]
Oh, great! I was working in this some time ago and I was using `https://github.com/sindresorhus/file-url` for files and used that in the `href`.

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: > Apply custom CSS to the webpage.

File: stream.js [hunk 2]
```diff
@@ -75,7 +75,15 @@ page.open(opts.url, function (status) {
 		if (!bgColor || bgColor === 'rgba(0, 0, 0, 0)') {
 			document.body.style.backgroundColor = 'white';
 		}
-	});
+
+		if (css) {
+			var style = document.createElement('style');
+			style.type = 'text/css';
```
Comment (sindresorhus) [inline stream.js]: Moot as it's the default.

**Comment by sindresorhus** [comment]
> Should screenshot-stream support a css file as well then?

Sure.

**Comment by SamVerschueren** [comment]
Added support for a css file but it's not working. Commented out the test. Anyone has an idea? Doesn't phantom support loading local files? Should I read the content of the file and just add it in the style tag?

**Comment by SamVerschueren** [comment]
Didn't seem to work with a local file in phantomjs. So now I read the file and inject the content in the page.

File: index.js [hunk 3]
```diff
@@ -49,6 +49,10 @@ module.exports = function (url, size, opts) {
 		es5shim = fs.readFileSync(es5Shim, 'utf8');
 	}
 
+	if (/\.css$/.test(opts.css)) {
+		opts.css = fs.readFileSync(opts.css, 'utf8');
```
Comment (SamVerschueren) [inline index.js]: Is this ok? Or should I use `path.join(process.cwd(), opts.css)`.

File: readme.md [hunk 4]
```diff
@@ -64,6 +64,12 @@ Type: `string`
 
 Capture a specific DOM element.
 
+##### css
+
+Type: `string`
+
+Apply custom CSS to the webpage.
```
Comment (SamVerschueren) [inline readme.md]: What about the description? Should we add something regarding file support and raw css support?

File: index.js (hunk 3 above)
Comment (sindresorhus) [inline index.js]: It's ok. `fs` does that internally already.

File: readme.md (hunk 4 above)
Comment (sindresorhus) [inline readme.md]: > Apply custom CSS to the webpage. Specify some CSS or the path to a CSS file.

**Comment by SamVerschueren** [comment]
Fixed description and merge conflict.

Will do a PR in `pageres` if this one lands to add a `css` method.

**Comment by SamVerschueren** [comment]
ping @kevva @sindresorhus :)

**Pull request closed without merging**